packages/components/src/range-control/rail.js

```diff
@@ -1,13 +1,9 @@
-/**
- * WordPress dependencies
- */
-import { isRTL } from '@wordpress/i18n';
```

Comment:
I think we should stay consistent in our isRTL test and only use the i18n one. If that check is broken, there are most likely other things (aside the range control) that are broken as well in the experience.

@sirreal and I worked on a fix for that isRTL in Gutenberg a few releases ago and I believe @noisysocks worked on backporting that fix to Core. 

So  I'd like to understand why it still doesn't work as intended.

Reply:
In other words, the direction in the document and the value returned by the i18n isRTL call should be equivalent, so we should try to figure out why it's not the case here.

Reply:
@swissspidy was also involved in these efforts lately as well.

Reply:
RTL Tester modifies `$wp_locale->text_direction`, which by default is set by the `_x( 'ltr', 'text direction' )` translation.

So any direction checks should probably just look at `$wp_locale->text_direction`

Reply:
I provided more screenshots were the `isRTL()` from `i18n` isn't working correctly when it's purely a `document.dir = 'rtl'` change:

https://github.com/WordPress/gutenberg/issues/29519#issuecomment-789938793

Reply:
So the issue is actually not in the coded but in the RTL tester plugin right?

Reply:
> RTL Tester modifies $wp_locale->text_direction, which by default is set by the _x( 'ltr', 'text direction' ) translation.

In the JS the source of truth is `_x( 'ltr', 'text direction' )` so if the RTL tester modifies just ` $wp_locale->text_direction` and not the result of the translation, it's not going to work properly in Gutenberg.

Reply:
So why isn't `$wp_locale->text_direction` the source of truth?

Reply:
FWIW, the RTL Tester plugin has been unchanged for years and working fine with `$wp_locale->text_direction`. That it now doesn't work anymore sounds like a backwards compatibility break IMO.

Reply:
That doesn't mean the issue is in the code, Gutenberg brought a lot of JS and JS translations which were not really a thing before. So I think we should consider a fix there. 

There's a possibility of making the translation of the JS based on the value of `$wp_locale->text_direction` but that sounds like a weird cycle dependency since the value of `$wp_locale->text_direction` depend on the backend translation.

Regardless, it seems to me that it something we can fix regardless of the timeline of the 5.7 since the issue is not on the production code.

Reply:
> There's a possibility of making the translation of the JS based on the value of $wp_locale->text_direction but that sounds like a weird cycle dependency since the value of $wp_locale->text_direction depend on the backend translation.

I'd just do `isRTL = 'rtl' === $wp_locale->text_direction`. No cycle.

Reply:
Gutenberg packages are packages that are independent of WordPress backend and should stay as is. The way we define whether a locale is RTL or not is by relying on the `ltr` translation (just like the backend), the  issue is that the backend has an intermediary value ($wp_locale->text_direction) that itself relies on the translation but the RTL plugin is tweaking the middle value instead of tweaking the source of truth.

Reply:
What I am saying is that `$wp_locale->text_direction` could be used here:

https://github.com/WordPress/wordpress-develop/blob/8ab1912b10eb1bd4bcf73f340b0aaee97a6c3fda/src/wp-includes/script-loader.php#L281


```php
if ( 'wp-i18n' === $handle ) {
	$script = sprintf( "wp.i18n.setLocaleData( { 'text direction\u0004ltr': [ '%s' ] } );", $wp_loccale->text_direction );
	$scripts->add_inline_script( $handle, $script, 'after' );
}

Reply:
Yeah, this is also what I suggested here https://github.com/WordPress/gutenberg/pull/29522#discussion_r586721621 but for me it doesn't seem right because it's a translation based on a variable that is itself based on the translation. 

It seems to me that we're changing valid production code to accommodate a test code. In general, I favor the opposite and adapt the test to the production code unless conceptually, the production code is broken but it doesn't seem to be the case for me here.

For instance, what happens if a php plugin relies on `_x( 'ltr', 'text direction' )`, it will also be broken with the RTL test plugin.

Reply:
Perhaps the isRTL function shouldn‘t have been depending on a translation in the first place, but a config flag.

Plugins are encouraged to use WP_Locale and is_rtl() instead of using core translations.

Reply:
Having an abstraction like `WP_Locale` class is fine and can be a good thing, what's problematic to me is the fact that you can modify the value of `$wp_locale->text_direction`.

We do have an abstraction/config like that in the frontend, it's just not a class, is `wp.i18n.isRTL` function.